projects/components/src/time-range/time-range.component.ts

```diff
@@ -90,7 +90,14 @@ export class TimeRangeComponent {
   @Output()
   public readonly timeRangeSelected: EventEmitter<TimeRange> = new EventEmitter();
 
-  public constructor(private readonly timeRangeService: TimeRangeService, private readonly ngZone: NgZone) {}
+  public constructor(
+    private readonly timeRangeService: TimeRangeService,
+    private readonly ngZone: NgZone,
+    private readonly refreshButtonService: RefreshButtonService,
+    private readonly subs: SubscriptionLifecycle
+  ) {
+    this.subs.add(this.refreshButtonService.refresh$.subscribe(() => this.timeRangeService.refresh()));
```

Comment:
so what benefit do we get from the service vs component output approach?

Reply:
Do you think I should just work with output then? Add an extra output in the **PageHeader** and let the consumer listen to it?

My initial thought was to have the service inside the timerange itself, but that won't be a good idea I think.

Reply:
Using event emitter instead of service.